### DISCUSSION TITLE
tests

### DISCUSSION THREAD

**Discussion by joshualerickson** [joshualerickson/exploreRGEE#7] [Ideas]
Hey @zackarno I think we should add some unit testing to these functions as well. I think it will help when we add/change stuff between us. Thoughts?

**Comment by zackarno** [comment]
Sounds like a good idea. Perhaps testing with continuous integration through github actions which looks to be already set up? I have been meaning to incorporate unit testing into more of my work for a while, but still have not done a great deal. If you have a couple test ideas that you can add I think that will help point me in the right direction

**Reply by joshualerickson** [reply]
I think just starting with the band and filter functions will be good. I'll add some later this week (hopefully) when I get around to the filter conversions. A good place to look is [rgee tests](https://github.com/r-spatial/rgee/blob/master/tests/testthat/test-ee_as_sf.R) for some ideas. When we start converting functions like (band -> reduce_zonal_ts) and others, it will be nice to run some simple checks before we commit to main. Have a good one :)

**Reply by zackarno** [reply]
cool cool thanks for the link. I'll start to incorporating some!
